### PR TITLE
MAID-2099: refactor/config: update config-handling

### PR DESCRIPTION
**Pull request by Fraser999**
This adds a new 'min_section_size' dev config option, and the Vault 'allow_multiple_lan_nodes' dev
option has been moved to Routing in this commit too.

The mock-crust Node and Client now make use of the 'min_section_size' config option rather than
passing this via the service handle.

Finally, a test for the sample config file has been added.

### CODE REVIEW

File: src/states/node.rs [hunk 1]
```diff
@@ -1995,8 +1995,16 @@ impl Node {
 
     /// Informs our peers that our section's member list changed. If `dst_prefix` is `Some`, only
     /// tells that section, otherwise tells all connected sections, including our own.
```
Comment (maqi) [inline src/states/node.rs]: shall we have a comment here for the new `allow_small_sections`?

File: tests/mock_crust/utils.rs [hunk 2]
```diff
@@ -62,6 +62,14 @@ pub fn gen_range_except<T: Rng>(
     x
 }
 
+fn create_config(network: &Network<PublicId>) -> Config {
+    Config {
+        dev: Some(DevConfig {
+            min_section_size: Some(network.min_section_size()),
+            ..DevConfig::default()
```
Comment (ustulation) [inline tests/mock_crust/utils.rs]: This `..` prefix is new to me - what does it mean ?

File: src/rate_limiter.rs [hunk 3]
```diff
@@ -182,45 +188,51 @@ mod tests {
...
-        let payload = unwrap!(serialisation::serialise(
-            &UserMessage::Request(Request::GetIData {
-                name: rand::random(),
-                msg_id: MessageId::new(),
-            }),
-        ));
-        let hash = sha3_256(&payload);
-        (payload, hash)
-    }
+    static LARGE_MESSAGE: [u8; CLIENT_CAPACITY as usize] = [0; CLIENT_CAPACITY as usize];
```
Comment (ustulation) [inline src/rate_limiter.rs]: Does this have to ba static instead of a const ?

File: tests/mock_crust/utils.rs (hunk 2 above)
Comment (Fraser999) [inline tests/mock_crust/utils.rs]: Means just fill in the rest of the uninitialised fields with those from the thing after the `..`.

File: src/rate_limiter.rs (hunk 3 above)
Comment (Fraser999) [inline src/rate_limiter.rs]: I was just trying to save resources slightly and avoid having a bunch of duplicated large arrays (which I think would happen if we used `const` instead, right?)

Comment (ustulation) [inline src/rate_limiter.rs]: hmm - interesting and also sounds correct to me :+1:

File: src/states/node.rs (hunk 1 above)
Comment (Fraser999) [inline src/states/node.rs]: Well there was a comment added to `RoutingTable::check_invariant()` which explains it OK I guess?  Or do you think we should still add a bit more info here too?

**Pull request merged**